### PR TITLE
workaround crash on exit

### PR DESCRIPTION
**Pull request by AndrewMeadows**

### CODE REVIEW

File: interface/src/main.cpp [hunk 1]
```diff
@@ -273,14 +273,14 @@ int main(int argc, const char* argv[]) {
...
 
-    qCDebug(interfaceapp, "Normal exit.");
+        qCDebug(interfaceapp, "Normal exit.");
 #if !defined(DEBUG) && !defined(Q_OS_LINUX)
-    // HACK: exit immediately (don't handle shutdown callbacks) for Release build
-    _exit(exitCode);
+        // HACK: exit immediately (don't handle shutdown callbacks) for Release build
+        _exit(exitCode);
 #endif
+    }
```
Comment (AndrewMeadows) [inline interface/src/main.cpp]: The trick here is to go straight to `_exit()` before the `~Application` dtor is invoked.

**Comment by AndrewMeadows** [comment]
The repro recipe that was working for me on Windows:

**(1)** Login to "SteamedHams".
**(2)** Spin the avatar's view around one revolution to load the rest of the scene.
**(3)** Press the **[X]** button to logout.
--> crash on shutdown (in RC65.1)
--> does not crash on shutdown

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8137/app_12681_PR.apk

File: interface/src/main.cpp (hunk 1 above)
Comment (Atlante45) [reply interface/src/main.cpp]: @AndrewMeadows we can't do that, there are a lot of important things tied to the Application's dtor.
One of them being writing the settings to the settings file.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8137/HighFidelity-Beta-PR12681-f398b9629165000d1aa942d98963e57603b7aa02.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8137/HighFidelity-Beta-PR12681-f398b9629165000d1aa942d98963e57603b7aa02.exe

**Comment by qlimnavarro** [comment]
Fail @conklin94122 
We are now seeing that when the domain is loading, the defect still occurs(crashes when closing via X). While loading local sandbox and SteamedHams, interface was closed and the "interface has stopped responding" windows message appeared.

In addition to this, when the domain is fully loaded and the close button is selected the interface does not crash. However on the next launch the "Interface crashed last time" window is presented. This is 10/10 reproducible and occurs on all systems (Mac OS, Windows 10, both desktop and HMD's)

Logs of mac and windows attached:
[Files.zip](https://github.com/highfidelity/hifi/files/1826901/Files.zip)

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8150/app_12681_PR.apk

File: interface/src/main.cpp (hunk 1 above)
Comment (Atlante45) [reply interface/src/main.cpp]: The teardown won't fix the issue.
The settings are cleaned up using `qAddPostRoutine` which runs a function inside the `QCoreApplication` dtor.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8150/HighFidelity-Beta-PR12681-1db2ada2ca10e1be959ca3103f880503ed33016f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8150/HighFidelity-Beta-PR12681-1db2ada2ca10e1be959ca3103f880503ed33016f.exe

**Comment by AndrewMeadows** [comment]
The "Interface crashed last time" problem is fixed, but the occasional "crash on X during scene load" problem still exists.

The problem appears to be hairy: I don't see an easy fix or workaround: we may need to dig into QT itself, or very carefully evaluate how we interact with it.  BTW, `hifi/master` also crashes on X during scene load in different but equally spectacular ways.

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Fail @conklin94122 
On Mac OS the interface crashes 6/10 times when selecting the X to close. 

On Windows 10 with desktop and HMD the interface (under the right conditions such as domain is fully loaded) does not crash and does not display the "interface crashed last time" message. On the minspec machine there was 1 crash out 10 that occurred, logs for that are also attached.

Logs for Mac OS:
[hifi-log 2.txt](https://github.com/highfidelity/hifi/files/1829778/hifi-log.2.txt)
[hifi-log 3.txt](https://github.com/highfidelity/hifi/files/1829779/hifi-log.3.txt)
[hifi-log 1.txt](https://github.com/highfidelity/hifi/files/1829780/hifi-log.1.txt)

Windows 10 Logs:
[4Logs.zip](https://github.com/highfidelity/hifi/files/1829861/4Logs.zip)

**Comment by conklin94122** [comment]
Spoke with Ryan. This fix is a significant improvement. We will merge it, discuss further work on the issue seperately.

**Pull request merged**